main.tf

```diff
@@ -3,6 +3,10 @@ resource "aws_wafv2_web_acl_association" "default" {
 
   resource_arn = var.association_resource_arns[count.index]
   web_acl_arn  = join("", aws_wafv2_web_acl.default.*.arn)
+
+  lifecycle {
+    ignore_changes = ignore_waf_associations ? [resource_arn, ] : []
```

Comment:
the `,` here is not necessary?

Reply:
it's not, but it doesn't hurt, TBF, the terraform docs has this same mistake

Reply:
actually you need to update this:

`ignore_changes = ignore_waf_associations ? [resource_arn, ] : [] `

needs to be:

`ignore_changes = var.ignore_waf_associations ? [resource_arn, ] : []`

Reply:
Actioning both of these, with no comma.